src/routers/__tests__/StackRouter-test.js

```diff
@@ -1007,6 +1007,126 @@ describe('StackRouter', () => {
     ]);
   });
 
+  test('Handles the navigate action with params and nested StackRouter as a first action', () => {
+    const state = TestStackRouter.getStateForAction({
+      type: NavigationActions.NAVIGATE,
+      routeName: 'main',
+      params: {
+        code: 'test',
+        foo: 'bar',
+      },
+      action: {
+        type: NavigationActions.NAVIGATE,
+        routeName: 'profile',
+        params: {
+          id: '4',
+          code: 'test',
+          foo: 'bar',
+        },
+        action: {
+          type: NavigationActions.NAVIGATE,
+          routeName: 'list',
+          params: {
+            id: '10259959195',
+            code: 'test',
+            foo: 'bar',
+          },
+        },
+      },
+    });
+
+    expect(state).toEqual({
+      index: 0,
+      isTransitioning: false,
+      key: 'StackRouterRoot',
+      routes: [
+        {
+          index: 0,
+          isTransitioning: false,
```

Comment:
The fact that these route objects has `isTransitioning` and `index` seemed weird to me. Seems like it happens because some code paths treat spread the state over the route, like https://github.com/react-navigation/react-navigation/blob/fae1f71fb3f3fcbf7ad8ffdb9e0f5f1a840bf660/src/routers/StackRouter.js#L123

I'm new to the codebase, so still groking things. Lmk if my suspicions are applicable.

Reply:
isTransitioning is a feature of the stack router. Each route in the stack may be a navigation state of another router, so we need to treat them as such, when the routeName matches